aspen/simplates/simplate.py

```diff
@@ -153,25 +154,32 @@ def render(self, accept, context):
                       over from the execution of the zeroth page
         """
 
+        # find matching media type
+        media_type = self.best_match(accept)
+        # create Output object and put it in the state
+        output = context['output'] = Output(media_type=media_type)
         # copy the state dict to avoid accidentally mutating it
         context = dict(context)
         # override it with values from the first page
         context.update(self.pages[0])
         # use this as the context to execute the second page in
         exec(self.pages[1], context)
+        # refetch output, this allows the second page to override it
+        output = context['state']['output']
+        # skip rendering if the second page has already filled output.body
+        if output.body:
```

Comment:
We should probably use `output.body is not None` here, so that the second page can force an empty body (`''`).